### ISSUE TITLE
Why's there JDK inside the camel-k-kit-* images?

### ISSUE THREAD

**Issue by bjethwan** [apache/camel-k#3424]
Why's there JDK inside the camel-k-kit-* images?
And 384MB image for a simple route with basic log and timer components ?

This is using one of your samples
```
cd bipin_projects/camel-k-example-basic
kamel run Basic.java
```
 
`dive bjethwan/camel-k-kit-cb2uqgpng1rc7386k9dg:6357`
<img width="1328" alt="image" src="https://user-images.githubusercontent.com/5045325/177846164-99a677f7-82f5-4de5-8bf6-460f7a68e600.png">

**Comment by astefanutti** [comment]
Thanks for raising it. JDK is required by the JOOR compiler, that compiles the Java routes at start time. It should be possible to use a JRE-only base image when using the other supported languages, and reduce the integration image size significantly.

**Comment by bjethwan** [comment]
@astefanutti I think you have a perfect case for cloud native buildpacks, paketo maybe.

**Comment by squakez** [comment]
We always need a JDK/JRE based container as at the end of the day we run a Camel application, so a buildpack style approach would be limited. Probably we can simply detect if an application has some java source (in which case we use the JDK), otherwise we can use a JRE image.

**Comment by squakez** [comment]
Closed in favour of #3743
